backend/src/index.ts

```diff
@@ -25,12 +30,15 @@ router.get('/chart/comment_threads/:id', getCommentThread);
 router.post('/chart/comment_threads', createCommentThread);
 router.post('/chart/comment_threads/:id/respond', respondToCommentThread);
 
-router.get('/share', getShareToken);
+router.get('/share/share_tokens', getAllShareTokens);
+router.post('/share/share_tokens', createShareToken);
+router.get('/share/share_tokens/:token', getShareToken);
 router.get('/chart/shared/:token', getSharedData);
 
 const handleErrors: ErrorRequestHandler = (error, req, res, next) => {
   if (error instanceof HttpError) {
     res.status(error.status).send(error.message);
+    return next();
```

Comment:
Fixes an issue where the error handler tries to send 500 errors for HTTP errors that are already handled.